libdevcore/CommonData.cpp

```diff
@@ -29,26 +29,49 @@
 using namespace std;
 using namespace dev;
 
+static char const* upperHexChars = "0123456789ABCDEF";
+static char const* lowerHexChars = "0123456789abcdef";
+
+string dev::toHex(uint8_t _data, HexCase _case)
+{
+	assertThrow(_case != HexCase::Mixed, BadHexCase, "Mixed case can only be used for byte arrays.");
+
+	char const* chars =
+		_case == HexCase::Upper ? upperHexChars : lowerHexChars;
+
+	return std::string{
+		chars[(unsigned(_data) / 16) & 0xf],
+		chars[unsigned(_data) & 0xf]
+	};
+}
+
 string dev::toHex(bytes const& _data, HexPrefix _prefix, HexCase _case)
 {
-	std::ostringstream ret;
+	std::string ret(_data.size() * 2 + (_prefix == HexPrefix::Add ? 2 : 0), 0);
+
+	size_t i = 0;
 	if (_prefix == HexPrefix::Add)
-		ret << "0x";
+	{
+		ret[i++] = '0';
+		ret[i++] = 'x';
+	}
 
 	int rix = _data.size() - 1;
 	for (uint8_t c: _data)
 	{
 		// switch hex case every four hexchars
-		auto hexcase = std::nouppercase;
+		char const* chars = lowerHexChars;
 		if (_case == HexCase::Upper)
```

Comment:
Shouldn't the ``Upper`` and ``Lower`` case be handled outside the loop and only the ``Mixed`` case inside the loop?

Reply:
Not that it matters too much...

Reply:
Sounds good!